app/models/spina/navigation_item.rb

```diff
@@ -11,8 +11,40 @@ class NavigationItem < ApplicationRecord
     scope :in_menu, -> { joins(:page).where(spina_pages: {show_in_menu: true}) }
     scope :active, -> { joins(:page).where(spina_pages: {active: true}) }
 
-    validates :page, uniqueness: {scope: :navigation}
+    validates :page, uniqueness: {scope: :navigation, allow_nil: true}
+    validate :url_and_url_label_presence
+    validate :url_or_page_presence
 
-    delegate :menu_title, :materialized_path, :draft?, :homepage?, to: :page
+    delegate :draft?, :homepage?, to: :page
```

Comment:
I think these also need their own method now, defaulting to false when there's no page. We might be able to get away with `allow_nil: true` here.

Reply:
Good call! That was an oversight on my part. Went ahead and added the `allow_nil: true` to take care of this.